internal/engine/fswatch/watchmanager.go

```diff
@@ -48,9 +53,94 @@ type WatchableTarget interface {
 
 var _ WatchableTarget = model.ImageTarget{}
 var _ WatchableTarget = model.LocalTarget{}
+var _ WatchableTarget = model.DockerComposeTarget{}
+
+func absPath(path string) (string, error) {
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return "", fmt.Errorf("could not determine absolute path for %q: %v", path, err)
+	}
+	return absPath, nil
+}
+
+func specForTarget(ctx context.Context, mn model.ManifestName, t WatchableTarget, globalIgnores []model.Dockerignore, dispatcher dispatcher) filewatches.FileWatchSpec {
+	logValidationError := func(err error) {
+		dispatcher.Dispatch(store.NewLogAction(
```

Comment:
hmmm...what prevents this from going into an infinite loop and logging the error on every change?

Reply:
Gahh, very valid point. Will restructure to prevent spamming on every store change to be only each time spec changes.